torch/optim/optimizer.py

```diff
@@ -1018,6 +1031,27 @@ def add_param_group(self, param_group: Dict[str, Any]) -> None:
         else:
             param_group["params"] = list(params)
 
+        extracted_param_tensors = []
+        param_group['param_names'] = []
+        for param in param_group["params"]:
+            if isinstance(param, tuple):
+                param_name = param[0]
+                param_group['param_names'].append(param_name)
+                extracted_param_tensors.append(param[1])
+            else:
+                extracted_param_tensors.append(param)
+                if len(param_group['param_names']) > 0:
+                    param_group['param_names'] = []
+                    extracted_param_tensors = []
+                    warnings.warn(
+                        "not all parameters in optimizer's parameter group are named_parameters. name is missing."
```

Comment:
I'm inclined to not want to allow this case at all--in other words, I think we should just error without affecting the optimizer state.